File: docs/examples/python/output/parts/UserManagementService_find_by_id.py
# code2map fragment (non-buildable)
# original: docs/examples/python/user_management_service.py
# lines: 139-149
# symbol: UserManagementService#find_by_id
# notes: references dataclasses.dataclass, re, typing.Optional; calls self._users.get
    def find_by_id(self, user_id: str) -> Optional[User]:
        """
        ユーザーIDでユーザーを検索する

        Args:
            user_id: 検索対象のユーザーID

        Returns:
            見つかったユーザー（見つからない場合はNone）
        """
        return self._users.get(user_id)

File: docs/examples/v0.1.2/java/output/parts/UserManagementService_findById.java
// code2map fragment (non-buildable)
// id: CD5
// original: docs/examples/java/UserManagementService.java
// lines: 110-112
// symbol: UserManagementService#findById
// notes: references java.util.ArrayList, java.util.HashMap, java.util.List, java.util.Map, java.util.Optional, java.util.regex.Pattern; calls Optional.ofNullable, users.get
    public Optional<User> findById(String userId) {
        return Optional.ofNullable(users.get(userId));
    }
